cpp/src/parquet/encoding_benchmark.cc

```diff
@@ -226,57 +226,30 @@ static void BM_ByteStreamSplitEncode(benchmark::State& state, EncodeFunc&& encod
-      state, arrow::util::internal::ByteStreamSplitDecodeScalar<float>);
+static void BM_ByteStreamSplitDecode_Float(benchmark::State& state) {
+  BM_ByteStreamSplitDecode<float>(state,
+                                  arrow::util::internal::ByteStreamSplitDecode<float>);
 }
 
-static void BM_ByteStreamSplitDecode_Double_Scalar(benchmark::State& state) {
-  BM_ByteStreamSplitDecode<double>(
-      state, arrow::util::internal::ByteStreamSplitDecodeScalar<double>);
+static void BM_ByteStreamSplitDecode_Double(benchmark::State& state) {
+  BM_ByteStreamSplitDecode<double>(state,
+                                   arrow::util::internal::ByteStreamSplitDecode<double>);
 }
 
-static void BM_ByteStreamSplitEncode_Float_Scalar(benchmark::State& state) {
-  BM_ByteStreamSplitEncode<float>(
-      state, arrow::util::internal::ByteStreamSplitEncodeScalar<float>);
+static void BM_ByteStreamSplitEncode_Float(benchmark::State& state) {
+  BM_ByteStreamSplitEncode<float>(state,
+                                  arrow::util::internal::ByteStreamSplitEncode<float>);
 }
 
-static void BM_ByteStreamSplitEncode_Double_Scalar(benchmark::State& state) {
-  BM_ByteStreamSplitEncode<double>(
-      state, arrow::util::internal::ByteStreamSplitEncodeScalar<double>);
+static void BM_ByteStreamSplitEncode_Double(benchmark::State& state) {
+  BM_ByteStreamSplitEncode<double>(state,
+                                   arrow::util::internal::ByteStreamSplitEncode<double>);
 }
 
-BENCHMARK(BM_ByteStreamSplitDecode_Float_Scalar)->Range(MIN_RANGE, MAX_RANGE);
-BENCHMARK(BM_ByteStreamSplitDecode_Double_Scalar)->Range(MIN_RANGE, MAX_RANGE);
-BENCHMARK(BM_ByteStreamSplitEncode_Float_Scalar)->Range(MIN_RANGE, MAX_RANGE);
-BENCHMARK(BM_ByteStreamSplitEncode_Double_Scalar)->Range(MIN_RANGE, MAX_RANGE);
-
-#if defined(ARROW_HAVE_SSE4_2)
-static void BM_ByteStreamSplitDecode_Float_SSE2(benchmark::State& state) {
-  BM_ByteStreamSplitDecode<float>(
-      state, arrow::util::internal::ByteStreamSplitDecodeSSE2<float>);
-}
-
-static void BM_ByteStreamSplitDecode_Double_SSE2(benchmark::State& state) {
-  BM_ByteStreamSplitDecode<double>(
-      state, arrow::util::internal::ByteStreamSplitDecodeSSE2<double>);
-}
-
-static void BM_ByteStreamSplitEncode_Float_SSE2(benchmark::State& state) {
-  BM_ByteStreamSplitEncode<float>(
-      state, arrow::util::internal::ByteStreamSplitEncodeSSE2<float>);
-}
-
-static void BM_ByteStreamSplitEncode_Double_SSE2(benchmark::State& state) {
-  BM_ByteStreamSplitEncode<double>(
-      state, arrow::util::internal::ByteStreamSplitEncodeSSE2<double>);
-}
-
-BENCHMARK(BM_ByteStreamSplitDecode_Float_SSE2)->Range(MIN_RANGE, MAX_RANGE);
-BENCHMARK(BM_ByteStreamSplitDecode_Double_SSE2)->Range(MIN_RANGE, MAX_RANGE);
-BENCHMARK(BM_ByteStreamSplitEncode_Float_SSE2)->Range(MIN_RANGE, MAX_RANGE);
-BENCHMARK(BM_ByteStreamSplitEncode_Double_SSE2)->Range(MIN_RANGE, MAX_RANGE);
-#endif
+BENCHMARK(BM_ByteStreamSplitDecode_Float)->Range(MIN_RANGE, MAX_RANGE);
```

Comment:
Why did you remove the different implementations? It becomes more difficult to compare them on a given CPU.